common/ontology.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import random
+import time
+
+from ontoma.interface import OnToma
+from pandarallel import pandarallel
+from pyspark.sql.functions import col, when
+
+ONTOMA_MAX_ATTEMPTS = 3
+pandarallel.initialize()
+
+
+def _simple_retry(func, **kwargs):
+    """Simple retry handling for functions. Cannot be a decorator, so that the functions could still be pickled."""
+    for attempt in range(1, ONTOMA_MAX_ATTEMPTS + 1):
+        try:
+            return func(**kwargs)
+        except:
+            # If this is not the last attempt, wait until the next one.
+            if attempt != ONTOMA_MAX_ATTEMPTS:
+                time.sleep(5 + 10 * random.random())
+    logging.error(f'OnToma lookup failed for {kwargs!r}')
+    return []
+
+
+def _ontoma_udf(row, ontoma_instance):
+    """Try to map first by disease name (because that branch of OnToma is more stable), then by disease ID."""
+    disease_name, disease_id = row['diseaseFromSource'], row['diseaseFromSourceId']
+    mappings = []
+    if disease_name:
+        mappings = _simple_retry(ontoma_instance.find_term, query=disease_name, code=False)
+    if not mappings and disease_id and ':' in disease_id:
```

Comment:
This way of looking for an ontology prefix won't work for all sources. For example, Orphanet's are represented with underscores (`Orphanet_2301`).

I suggest translating all underscores to a colon for now and when the bugs are fixed at the OnToma level, manually account for all the different ontology prefixes.

Reply:
You're right, thank you for pointing it out.  Once we fix OnToma properly, this problem will disappear, because OnToma already performs identifier normalisation internally